cmake/FindLibWebSockets.cmake

```diff
@@ -45,10 +45,11 @@ find_path(LIBWEBSOCKETS_INCLUDE_DIRS
   PATHS "/usr/include"
   )
 
+# strips trailing version elaboration, e.g. #define LWS_LIBRARY_VERSION "4.1.6-git..."
 if(LIBWEBSOCKETS_INCLUDE_DIRS AND EXISTS "${LIBWEBSOCKETS_INCLUDE_DIRS}/lws_config.h")
   file(STRINGS "${LIBWEBSOCKETS_INCLUDE_DIRS}/lws_config.h" lws_version_str
     REGEX "^#define[ \t]+LWS_LIBRARY_VERSION[ \t]+\"[^\"]+\"")
-  string(REGEX REPLACE "^#define[ \t]+LWS_LIBRARY_VERSION[ \t]+\"([^\"]+)\".*" "\\1"
+  string(REGEX REPLACE "^#define[ \t]+LWS_LIBRARY_VERSION[ \t]+\"([0-9.]+).*" "\\1"
```

Comment:
This is the cause that the test was not being run on CentOS 7. There, the `LWS_LIBRARY_VERSION` ended up being `3.0.1-`.

Reply:
> This is the cause that the test was not being run on CentOS 7. There, the `LWS_LIBRARY_VERSION` ended up being `3.0.1-`.

Can we please add the above as a comment ? Thanks.

Reply:
I already added `# strips trailing version elaboration, e.g. #define LWS_LIBRARY_VERSION "4.1.6-git..."`. Line 48. Should be sufficient to explain possible problem with `-`, imo.